sgx/sgx_mock_test.go

```diff
@@ -0,0 +1,41 @@
+package sgx_test
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/zama-ai/fhevm-go/fhevm/tfhe"
+	"github.com/zama-ai/fhevm-go/sgx"
+	"pgregory.net/rapid"
```

Comment:
One thing is that when these tests fail, it will output some files on why it failed (e.g. with which random values). These files should not be committed to git, so let's add to .gitignore:

```
**/testdata/rapid
```

You can test this by adding a `t.Fatal("force fail test")` at the end of the test